charms/worker/k8s/tests/unit/test_k8sd_api_manager.py

```diff
@@ -262,13 +263,29 @@ def test_update_cluster_config(self, mock_send_request):
 
         dns_config = DNSConfig(enabled=True)
         user_config = UserFacingClusterConfig(dns=dns_config)
-        request = UpdateClusterConfigRequest(config=user_config)
+        datastore = UserFacingDatastoreConfig(
+            type="external",
+            servers=["localhost:123"],
+            ca_crt="ca-crt",
```

Comment:
for some reason, this fails. I set the alias for those fields but it will only accept `ca-crt` and ignores `ca_crt`?

Reply:
What's the problem here? I think I don't follow... Did you mean that the model is serialized using `ca_crt` as the JSON key instead of `ca-crt`.

Reply:
I see it now: 
```
AssertionError: expected call not found.
Expected: _send_request('/1.0/k8sd/cluster/config', 'PUT', <class 'lib.charms.k8s.v0.k8sd_api_manager.EmptyResponse'>, {'config': {'dns': {'enabled': True}}, 'datastore': {'type': 'external', 'servers': ['localhost:123'], 'ca-crt': 'ca-crt', 'client-crt': 'client-crt', 'client-key': 'client-key'}})
Actual: _send_request('/1.0/k8sd/cluster/config', 'PUT', <class 'lib.charms.k8s.v0.k8sd_api_manager.EmptyResponse'>, {'config': {'dns': {'enabled': True}}, 'datastore': {'type': 'external', 'servers': ['localhost:123']}})
```

Reply:
yes, `allow_population_by_field` was the solution, thanks!